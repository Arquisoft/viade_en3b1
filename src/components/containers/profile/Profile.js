import React, {Component} from 'react';
import { LoggedIn, LoggedOut, Value, List } from '@solid/react';

class Profile extends Component {  
    render() {

        const childrenTrimmed = (item, index) =>
            <li key={index}>
                <a href = {item}>
                    {`${item}`.split('.')[0].split('//')[1]}
                </a>
            </li>;

        return (
            <div>
                <h2>Your profile</h2>
                <p>
                    <LoggedOut>You are here as a guest.</LoggedOut>
                    <LoggedIn>Logged in as <Value src="user.name"/>
                    <ul>
                        <li>Name:   <Value src="user.vcard_fn"/></li>
                        <li>Friends: {'$user.vcard_url'}
                            <List src = "user.vcard_url"
                                children = {childrenTrimmed} >
                            </List>
                        </li>
                    </ul>
                    </LoggedIn>  
                </p>
            </div>
        );
    }
}

export default Profile;