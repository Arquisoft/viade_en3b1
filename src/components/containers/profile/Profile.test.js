import Profile from './Profile';

test('Define Profile', () => {
  expect(Profile).toBeDefined();
});

test('Renders correctly', () => {
  const wrapper = render(<Profile />);
  expect(wrapper).toMatchSnapshot();
});