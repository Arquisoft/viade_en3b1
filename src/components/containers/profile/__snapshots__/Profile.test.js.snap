// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  class="makeStyles-root-1"
>
  <div
    class="makeStyles-root-8"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-9 MuiToolbar-gutters"
      >
        <div
          class="MuiGrid-root makeStyles-link-11"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorInherit"
              href="#/home"
            >
              <div
                class="MuiGrid-root MuiGrid-container"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <img
                    alt="Viade Logo"
                    src="logo_2.svg"
                    style="height:2.5rem;width:2.5rem"
                  />
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <h4
                    class="MuiTypography-root MuiTypography-h4"
                  >
                    ViaDe
                  </h4>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </header>
  </div>
  <main
    class="MuiContainer-root makeStyles-main-2 MuiContainer-maxWidthMd"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3"
    >
      Your Profile
    </h3>
    <div
      class="MuiPaper-root makeStyles-paper-3 MuiPaper-elevation0 MuiPaper-rounded"
    >
      <div
        class="MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed"
          style="overflow:hidden"
        >
          <div
            class="MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected"
              role="tab"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiTab-wrapper"
              >
                General
              </span>
              <span
                class="PrivateTabIndicator-root-226 PrivateTabIndicator-colorPrimary-227 MuiTabs-indicator"
              />
            </button>
            <a
              aria-disabled="false"
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary"
              href="#/profile/friends"
              role="tab"
              tabindex="0"
            >
              <span
                class="MuiTab-wrapper"
              >
                Friends
              </span>
            </a>
            <a
              aria-disabled="false"
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary"
              href="#/profile/groups"
              role="tab"
              tabindex="0"
            >
              <span
                class="MuiTab-wrapper"
              >
                Groups
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-4 MuiPaper-outlined MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle makeStyles-photo-5 MuiAvatar-colorDefault"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiAvatar-fallback"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-flexItem MuiDivider-vertical"
          />
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <h3
              class="MuiTypography-root makeStyles-name-6 MuiTypography-h3"
            >
              <span
                class="solid value pending"
              />
            </h3>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-4 MuiPaper-outlined MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1"
            >
              Need to change anything?
            </h6>
          </div>
          <h6
            class="MuiTypography-root makeStyles-link1-7 MuiTypography-subtitle1"
          >
            Go to your 
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="0"
              style="color:#7c4dff"
              target="_blank"
            >
              Solid account
            </a>
          </h6>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-4 MuiPaper-outlined MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1"
            >
              Or you can also 
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href=""
                style="color:#8693E3"
              >
                sign out
              </a>
              .
            </h6>
          </div>
        </div>
      </div>
    </div>
  </main>
  <div>
    <footer
      class="makeStyles-footer-257"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXs"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6"
            >
              ViaDe En3B1
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
            >
              Made by 
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="https://github.com/PabloCanalSuarez"
                style="color:#302c58"
                target="_blank"
              >
                Pablo Ca√±al
              </a>
            </h6>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-flexItem MuiDivider-vertical"
          />
          <div
            class="MuiGrid-root MuiGrid-item"
            style="margin-top:0.5rem"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              href="https://github.com/Arquisoft/viade_en3b1/"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
